version_check.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+import sys
+from typing import Optional, Dict
+import tempfile
+
+import mistune
+import CppHeaderParser
+import semver
+import git
+
+
+def die(*args, error_code=1, **kwargs):
+    print('ERROR: ', end='', file=sys.stderr)
+    print(*args, **kwargs, file=sys.stderr)
+    exit(error_code)
+
+
+class SemVerWithVPrefix(semver.VersionInfo):
+    @classmethod
+    def parse(cls, version):
+        if not version[0] in ('v', 'V'):
+            raise ValueError(f"{version}: not a valid semantic version tag. Must start with 'v' or 'V'")
+        self = super(SemVerWithVPrefix, cls).parse(version[1:])
+        return self
+
+    def __str__(self):
+        return 'v' + super(SemVerWithVPrefix, self).__str__()
+
+
+def get_header_defines(header_contents: str) -> Dict[str, str]:
+    header = CppHeaderParser.CppHeader(header_contents, argType='string')
+    defines_dict = {}
+    for d in header.defines_detail:
+        define_tup = d['value'].split(' ', maxsplit=1)
+        defines_dict[define_tup[0]] = define_tup[1]
+    return defines_dict
+
+
+def find_child_text(node) -> Optional[str]:
+    if 'children' in node.keys():
+        if isinstance(node['children'], list):
+            return find_child_text(node['children'][0])
+    elif 'text' in node.keys():
+        return node['text']
+    return None
```

Comment:
this code can fail if e.g. 'children' key exists and its value is an empty list. also you can simply use node.get('children') to get the value or None if key does not exist. Assuming that 'children' value is always a list (if existent), the code can be shortened to:

```python
def find_child_text(node: dict) -> Optional[str]:
    if node.get('children'):
        return find_child_text(node.get('children')[0])
    else:
        return node.get('text')
```

Reply:
I don't like the implicit assumption that `['children']` always contains a list. Mistune documentation is pretty sparse on it's AST generation (https://mistune.readthedocs.io/en/latest/guide.html?highlight=ast#astrenderer). I'll refactor a bit to fix the `[]` bug, maybe you'll like the new version better :smiley: